packages/block-library/src/navigation/edit.js

```diff
@@ -183,6 +176,13 @@ function Navigation( {
 
 export default compose( [
 	withSelect( ( select, { clientId } ) => {
+		const { getBlockSupport } = select( blocksStore );
+		const allowedBlocks = getBlockSupport(
+			'core/navigation',
+			'__experimentalAllowedBlocks',
+			[ 'core/navigation-link' ] // default to the basic link block only.
```

Comment:
Not sure this is required as the block.json already represents defaults for the block.

Usually the default for block supports is something that applies over multiple block types, like `reusable` defaulting to `true` for all blocks when not specified in a block's supports, but here this only applies to the navigation block, so probably not needed.